### PR TITLE
fix: escape special characters in Telegram command messages for MarkdownV2

### PR DESCRIPTION
**Pull request by JorgePasco1**
## Summary
- Fixed Telegram bot commands (/start, /subscribe, /unsubscribe, /status) failing with "Bad Request: can't parse entities" errors
- All command messages now properly escape MarkdownV2 special characters (! - .)
- Admin notification format also updated with proper escaping

## Root Cause
The `send_message` function uses MarkdownV2 parse_mode, but command response messages were plain text without escaping special characters like `!`, `-`, and `.`

## Test plan
- [x] All 367 existing tests pass
- [ ] Test /start command in production bot
- [ ] Verify other commands (/subscribe, /unsubscribe, /status) work correctly

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Ensured all Telegram messages use consistent MarkdownV2 escaping so special characters (including parentheses, punctuation, links, emojis and multi-line content) render correctly across commands, welcome messages, summaries and admin notifications.
  * Standardized message composition to prevent unescaped character regressions.

* **Tests**
  * Expanded unit, integration-like and regression tests covering per-character, sequence and edge-case escaping, payload serialization and command responses.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/telegram.rs [hunk 1]
```diff
@@ -2341,4 +2341,911 @@ Discussion on #AI trends and python_utils.py examples."#;
...
+
+        // Test that ) in URL gets escaped when detected
+        // Note: Current implementation has a limitation with nested parens in URLs
+        // This documents actual behavior
+        let text_with_paren_in_url =
+            "[Link](https://en.wikipedia.org/wiki/Rust_(programming_language\\))";
+        // This URL has a pre-escaped ) to work around the limitation
+        let escaped2 = escape_markdownv2(text_with_paren_in_url);
+        assert!(escaped2.contains("Rust_"));
+    }
```
[the remark this answers is not in the document]
Comment (JorgePasco1) [reply src/telegram.rs]: Yes please, create that github issue
